src/components/ProfileButton/ProfileButton.jsx

```diff
@@ -1,13 +1,32 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { FaUser } from 'react-icons/fa';
-import { Button } from './styled';
+import useGlobal from '../../hooks/useGlobal';
+import { Button, ToggleWrapper, ButtonWrapper, Option, NavLink, Avatar } from './styled';
 
 function ProfileButton() {
-  return (
-    <Button>
-      <FaUser />
-    </Button>
-  );
+  const [visible, setVisible] = useState(false),
+    { globalState, logout } = useGlobal();
+  let authenticatedOptions = <Option onClick={logout}>Logout</Option>,
+    guestOptions = (
+      <Option>
+        <NavLink to="/login">Login</NavLink>
+      </Option>
+    ),
+    profileButton = (
+      <ButtonWrapper onBlur={() => setVisible(false)}>
+        <Button onClick={() => setVisible(!visible)}>
+          {globalState.authenticated ? (
+            <Avatar src={globalState.userData.avatarUrl} alt="User avatar" />
+          ) : (
+            <FaUser />
+          )}
+        </Button>
+        <ToggleWrapper visible={visible}>
+          {globalState.authenticated ? authenticatedOptions : guestOptions}
+        </ToggleWrapper>
+      </ButtonWrapper>
+    );
+  return profileButton;
```

Comment:
I recommend returning the JSX here, it is not adding more readiness with the name of the variable